src/teaching-element/BrightcoveVideo.vue

```diff
@@ -66,13 +66,21 @@ export default {
       // Clear errors.
       this.error = null;
       // Remove player assets.
-      if (this.script) this.$el.removeChild(this.script);
-      this.script = null;
-      if (this.style) document.head.removeChild(this.style);
-      this.style = null;
+      this.removePlayerAssets();
       // Wipe wrapper contents.
       empty(this.$refs.videoWrapper);
     },
+    removePlayerAssets() {
+      let { script, style, $el } = this;
+      if (script && $el.contains(script)) {
+        $el.removeChild(script);
+        script = null;
```

Comment:
Once destructured all further changes that are made on it won't be reflected on source object.